extension/ReactWebview.ts

```diff
@@ -0,0 +1,119 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as v4 from 'uuid/v4';
+
+export class ReactWebView {
+	public static currentPanel: vscode.WebviewPanel | undefined;
+
+	constructor(extensionPath: string) {
+    const column = vscode.window.activeTextEditor
+      ? vscode.window.activeTextEditor.viewColumn
+      : undefined;
+
+		// Check for existing panel
+		if (ReactWebView.currentPanel) {
+			ReactWebView.currentPanel.reveal(column);
+		} else {
+			ReactWebView.currentPanel = vscode.window.createWebviewPanel(
+        'openRpc',
+        'Preview: openrpc.json',
+				vscode.ViewColumn.One,
+				{
+					enableScripts: true
+				}
+      );
+
+			ReactWebView.currentPanel.webview.html = this.getWebviewContent(extensionPath);
+
+			// Clean up resource when panel disposes
+			ReactWebView.currentPanel.onDidDispose(() => {
+				if (ReactWebView.currentPanel) {
+					ReactWebView.currentPanel.dispose();
+					ReactWebView.currentPanel = undefined;
+				}
+			});
+
+			vscode.workspace.findFiles('openrpc.json')
+				.then(files => {
+					const openRpcDoc = files[0]; // using the file at root
+
+					const docData = require(openRpcDoc.path);
+					this.updateContent(docData);
+				});
+
+			vscode.workspace.onDidChangeTextDocument(({ contentChanges, document }) => {
+				if (document.fileName.endsWith('openrpc.json') && !document.isDirty) {
```

Comment:
```suggestion
				if (!document.isDirty) {
```

Reply:
this gets taken care of in the [package.json#L25](https://github.com/open-rpc/editor-extensions-vscode/blob/master/package.json#L25)